ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -685,20 +693,29 @@ instance Condense (ChainHash (ByronBlockOrEBB cfg)) where
   condense (BlockHash h) = condense h
 
 instance Condense (GenTx (ByronBlockOrEBB cfg)) where
-    condense (ByronTx { byronTx = tx }) =
-      "(tx: "       <> show (CC.UTxO.aTaTx tx) <>
-      ", witness: " <> show (CC.UTxO.aTaWitness tx) <>
-      ")"
+    condense (ByronTx _ tx)            = T.unpack $ sformat build (void tx)
+    condense (ByronDlg _ cert)         = T.unpack $ sformat build (void cert)
+    condense (ByronUpdateProposal _ p) = T.unpack $ sformat build (void p)
+    condense (ByronUpdateVote _ vote)  = T.unpack $ sformat build (void vote)
 
 instance Show (GenTx (ByronBlockOrEBB cfg)) where
     show tx = condense tx
 
 instance Condense (GenTxId (ByronBlockOrEBB cfg)) where
-  condense = condense . unByronTxId
+  condense (ByronTxId i)             = condense i
+  condense (ByronDlgId i)            = condense i
+  condense (ByronUpdateProposalId i) = condense i
+  condense (ByronUpdateVoteId i)     = condense i
 
 instance Show (GenTxId (ByronBlockOrEBB cfg)) where
   show = condense
 
+instance Condense DlgCertId where
+  condense h = "DlgCertId: " <> T.unpack (sformat Crypto.shortHashF h)
+
+instance Condense UpdateVoteId where
```

Comment:
These two instances are orphans. Best to put them in `Ouroboros.Consensus.Util.Condense`.

Reply:
In this case, probably best that I move `DlgCertId` and `UpdateVoteId` to `cardano-ledger` since they're defined in this module.

Reply:
Doing this right now.

Reply:
Ok, but that doesn't solve the orphan problem, as `Condense` is not available in `cardano-ledger`.

Reply:
I should have explained further. These will remain orphans but be placed in `Ouroboros.Consensus.Util.Condense`. If I leave the definitions of `DlgCertId` and `UpdateVoteId` types in this module, there will be a cyclic dependency between `Ouroboros.Consensus.Util.Condense` and `Ouroboros.Consensus.Ledger.Byron`. I could factor the definitions out into another module here in `ouroboros-consensus`, but I felt like it makes more sense to just put them in `cardano-ledger`.

Reply:
It seems like we're on the same page after :slightly_smiling_face: